src/docker-images/gpu-reporter/reporter.py

```diff
@@ -135,50 +157,68 @@ def get_monthly_idleness(prometheus_url):
 
-        booked_seconds = values[-1][0] - values[0][0] + step_seconds
-        idleness_seconds = 0
-        nonidle_util_sum = 0.0
-
         for time, utils in values:
             utils = float(utils)
-            if utils <= IDLENESS_THRESHOLD:
-                idleness_seconds += step_seconds
-            else:
-                nonidle_util_sum += utils * step_seconds
-
-        vc_level[vc_name][username]["booked"] += booked_seconds
-        vc_level[vc_name][username]["idle"] += idleness_seconds
-        vc_level[vc_name][username]["nonidle_util_sum"] += nonidle_util_sum
-
-        user_level[vc_name][username][job_id]["booked"] += booked_seconds
-        user_level[vc_name][username][job_id]["idle"] += idleness_seconds
-        user_level[vc_name][username][job_id][
-            "nonidle_util_sum"] += nonidle_util_sum
-
-    for vc_name, vc_values in vc_level.items():
-        for username, user_val in vc_values.items():
-            nonidle_time = user_val["booked"] - user_val["idle"]
-            nonidle_util_sum = user_val["nonidle_util_sum"]
-
-            if nonidle_time == 0:
-                user_val["nonidle_util"] = 0.0
-            else:
-                user_val["nonidle_util"] = nonidle_util_sum / nonidle_time
-            user_val.pop("nonidle_util_sum")
-
-    for vc_name, vc_values in user_level.items():
-        for username, user_values in vc_values.items():
-            for job_id, job_val in user_values.items():
-                nonidle_time = job_val["booked"] - job_val["idle"]
-                nonidle_util_sum = job_val["nonidle_util_sum"]
+
+            for _, ago, vc_level in vc_levels:
+                if ago < time:
+                    continue
+
+                vc_level[vc_name][username]["booked"] += step_seconds
+                if utils <= IDLENESS_THRESHOLD:
+                    vc_level[vc_name][username]["idle"] += step_seconds
+                else:
+                    vc_level[vc_name][username][
+                        "nonidle_util_sum"] += utils * step_seconds
+
+            for _, ago, user_level in user_levels:
+                if ago < time:
+                    continue
+
+                user_level[vc_name][username][job_id]["booked"] += step_seconds
+                if utils <= IDLENESS_THRESHOLD:
+                    user_level[vc_name][username][job_id][
+                        "idle"] += step_seconds
+                else:
+                    user_level[vc_name][username][job_id][
+                        "nonidle_util_sum"] += utils * step_seconds
+
+    for _, _, vc_level in vc_levels:
+        for vc_name, vc_values in vc_level.items():
+            for username, user_val in vc_values.items():
+                nonidle_time = user_val["booked"] - user_val["idle"]
+                nonidle_util_sum = user_val["nonidle_util_sum"]
 
                 if nonidle_time == 0:
-                    job_val["nonidle_util"] = 0.0
+                    user_val["nonidle_util"] = 100.0
```

Comment:
If there is no non-idle, then all is idle, then nonidle_util should be zero?

Reply:
ok, I think I confused myself on this.